conanfile_base.py

```diff
@@ -0,0 +1,75 @@
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        sha256 = "e87aae032bf07c26f85ac0ed3250998c37621d95f8bd748b31f15b33c45ee995"
+        tools.get("{0}/releases/download/v{1}/{2}-{3}.tar.gz".format(self.homepage, self.version, self._base_name, self.version), sha256=sha256)
+        extracted_dir = self._base_name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        if tools.os_info.is_windows:
+            self.requires("pcre2/10.32@bincrafters/stable")
+
+    def build(self):
+        for filename in sorted(glob.glob("patches/*.patch")):
+            self.output.info('applying patch "%s"' % filename)
+            tools.patch(base_path=self._source_subfolder, patch_file=filename)
+
+        env_build = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        configure_args = ["--disable-nls"]
+        if "shared" in self.options and self.options.shared:
+            configure_args.extend(["--enable-shared", "--disable-static"])
+        else:
+            configure_args.extend(["--disable-shared", "--enable-shared"])
+
+        if str(self.settings.compiler) == "gcc" and float(str(self.settings.compiler.version)) >= 6:
+            configure_args.append("ac_cv_func_reallocarray=no")
+        with tools.chdir(self._source_subfolder):
+            if tools.os_info.is_windows:
+                tools.save("regex.h", "#define PCRE2_CODE_UNIT_WIDTH 8\n#include <pcre2posix.h>")
+                if not os.path.isdir("sys"):
+                    os.makedirs("sys")
+                tools.download("https://raw.githubusercontent.com/win32ports/sys_wait_h/master/sys/wait.h", os.path.join("sys", "wait.h"))
+                env_build.include_paths.append(os.getcwd())
+            if tools.cross_building(self.settings):
+                # stage1flex must be built on native arch: https://github.com/westes/flex/issues/78
+                self.run("./configure %s" % " ".join(configure_args))
+                env_build.make(args=["-C", "src", "stage1flex"])
+                self.run("mv src/stage1flex src/stage1flex.build")
+                env_build.make(args=["distclean"])
+                with tools.environment_append(env_build.vars):
+                    env_build.configure(args=configure_args)
+                    cpu_count_option = "-j%s" % tools.cpu_count()
+                    self.run("make -C src %s || true" % cpu_count_option)
+                    self.run("mv src/stage1flex.build src/stage1flex")
+                    self.run("touch src/stage1flex")
+                    env_build.make(args=["-C", "src"])
+            else:
+                with tools.environment_append(env_build.vars):
+                    env_build.configure(args=configure_args)
+                    env_build.make()
+            env_build.make(args=["install"])
+
+    def package(self):
```

Comment:
You will need to override package for each recipe. You can call this package method by super.
Or maybe, you could install all stuff and remove what you want.

Reply:
package only copies license, which is the same for both, what's the point on overriding it?

Reply:
make install probably will deploy flex binary, headers and lib, but we have one installer package, which only cares about the bin and the library package, which does't not require the bin. However, it's not possible (I think) choose what you want to install by configure arguments.

Reply:
the only thing we can to is to remove unwanted stuff in package() method, however, it doesn't appear to be necessary, as only `flex_installer`'s binary goes to the PATH

Reply:
I prefer removing if possible. I've removed the libraries and headers from here: https://github.com/uilianries/conan-flex-bincrafters/blob/testing/2.6.4/conanfile_installer.py#L21

Reply:
Done via https://github.com/bincrafters/conan-flex/pull/7/commits/46767dc19b784e68257b866488b7ef3ce7919692